### PR TITLE
LUCENE-9702: Hunspell: support alternate casing for short language codes

### PR DESCRIPTION
**Pull request by donnerpeter**
# Description

Turkish `I`s don't work with `LANG tr`

# Solution

Support `tr` as well as `tr_TR`

# Tests

`dotless_i` from Hunspell/C++

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `master` branch.
- [x] I have run `./gradlew check`.
- [x] I have added tests for my changes.
- [ ] I have added documentation for the [Ref Guide](https://github.com/apache/lucene-solr/tree/master/solr/solr-ref-guide) (for Solr changes only).

### CODE REVIEW

File: lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/Dictionary.java [hunk 1]
```diff
@@ -370,7 +370,7 @@ private void readAffixFile(InputStream affixStream, CharsetDecoder decoder)
         fullStrip = true;
       } else if ("LANG".equals(firstWord)) {
         language = singleArgument(reader, line);
-        alternateCasing = "tr_TR".equals(language) || "az_AZ".equals(language);
```
Comment (dweiss) [inline lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/Dictionary.java:373]: Hmm... I didn't check but can we use Locale to parse it properly to extract the language from country variant?

Comment (donnerpeter) [reply lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/Dictionary.java:373]: Probably we can, but `Locale`s javadoc is so scary and the class looks so general that I'd prefer this simple code.

**Comment by donnerpeter** [comment]
Force-pushed to resolve a conflict in `SpellCheckerTest.java`, other files haven't changed

File: lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/Dictionary.java (hunk 1 above)
Comment (dweiss) [reply lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/Dictionary.java:374]: Locale reflects what those iso codes are in the wild... That's why I'm a bit uneasy about using a simple prefix... It should at least be something like this: language.equalsIgnoreCase("tr") || language.toLowerCase(Locale.ROOT).startsWith("tr_")

Comment (donnerpeter) [reply lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/Dictionary.java:374]: Good point. So it seems that some parsing is inevitable, and quick googling shows that JDK doesn't seem to have that parsing. So I'll split by '_' myself.

Comment (dweiss) [reply lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/Dictionary.java:374]: I looked at that too - surprising there isn't iso -> locale conversion built-in. Sorry.

**Pull request merged**